File: internal/models/audit.go
package models

import "time"

type AuditEvent struct {
	Base
	OrganizationID *string   `json:"organizationId,omitempty" gorm:"type:uuid;index"`
	UserID         *string   `json:"userId,omitempty" gorm:"type:uuid;index"`
	Action         string    `json:"action" gorm:"not null;index"`
	ResourceType   string    `json:"resourceType" gorm:"not null"`
	ResourceID     string    `json:"resourceId,omitempty" gorm:"index"`
	IPAddress      string    `json:"ipAddress,omitempty"`
	UserAgent      string    `json:"userAgent,omitempty"`
	Metadata       string    `json:"metadata" gorm:"type:jsonb;not null;default:'{}'"`
	OccurredAt     time.Time `json:"occurredAt" gorm:"not null;index"`
}

File: apps/web/src/interfaces/audit.ts
import type { Entity } from '#/interfaces/api';

export type AuditEvent = Entity & {
  organizationId?: string;
  userId?: string;
  action: string;
  resourceType: string;
  resourceId?: string;
  ipAddress?: string;
  userAgent?: string;
  metadata: string;
  occurredAt: string;
};
